### PR TITLE
feat: floating interdomain and interdomain improvements and fixes after manual testing

### PR DESCRIPTION
**Pull request by denis-tingaikin**
Signed-off-by: Denis Tingaikin <denis.tingajkin@xored.com>

## Issue links
https://github.com/networkservicemesh/sdk/issues/853
https://github.com/networkservicemesh/sdk/issues/854
Closes https://github.com/networkservicemesh/sdk/issues/827

## How Has This Been Tested?

- [X] Added unit testing to cover
- [X] Tested manually
- [X] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [X] Bug fix
- [X] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/chains/nsmgrproxy/server_test.go [hunk 1]
```diff
@@ -99,5 +298,252 @@ func TestNSMGR_InterdomainUseCase(t *testing.T) {
...
+// Test_Interdomain_PassThroughUsecase covers scenario when we have 5 clusters.
+// Each cluster contains NSE with name endpoint-${cluster-num}.
+// Each endpoint request endpoint from the previous cluster (exclude 1st).
+//
+//
+// nsc -> nsmgr4 ->  forwarder4 -> nsmgr4 -> nsmgr-proxy4 -> nsmgr-proxy3 -> nsmgr3 ->forwarder3 -> nsmgr3 -> nse3 ->
+// nse3 -> nsmgr3 ->  forwarder3 -> nsmgr3 -> nsmgr-proxy3 -> nsmgr-proxy2 -> nsmgr2 ->forwarder2-> nsmgr2 -> nse2 ->
+// nse2 -> nsmgr2 ->  forwarder2 -> nsmgr2 -> nsmgr-proxy2 -> nsmgr-proxy1 -> nsmgr1 -> forwarder1 -> nsmgr1 -> nse1 ->
+// nse1 -> nsmgr1 ->  forwarder1 -> nsmg1 -> nsmgr-proxy1 -> nsmgr-proxy0 -> nsmgr0 -> forwarder0 -> nsmgr0 -> nse0
+
```
Comment (denis-tingaikin) [inline pkg/networkservice/chains/nsmgrproxy/server_test.go:477]: I think it would be nice to see this example working on real clusters :)

**Comment by denis-tingaikin** [comment]
@edwarnicke Would you like to split it into pieces?

File: pkg/tools/clienturlctx/context.go [hunk 2]
```diff
@@ -36,6 +38,7 @@ func WithClientURL(parent context.Context, clientURL *url.URL) context.Context {
 	if parent == nil {
 		panic("cannot create context from nil parent")
 	}
+	log.FromContext(parent).Infof("passed clientURL: %v", clientURL)
```
Comment (edwarnicke) [inline pkg/tools/clienturlctx/context.go:41]: Could we make this Debug not Info level?

File: pkg/registry/common/dnsresolve/common.go [hunk 3]
```diff
@@ -84,6 +92,7 @@ func resolveDomain(ctx context.Context, service, domain string, r Resolver) (*ur
 		return nil, err
 	}
 
+	log.FromContext(ctx).Info("Resolved url: %v", u)
```
Comment (edwarnicke) [inline pkg/registry/common/dnsresolve/common.go:95]: Could we make this Debug level?

File: pkg/tools/sandbox/builder.go [hunk 4]
```diff
@@ -269,21 +289,28 @@ func (b *Builder) dialContext(ctx context.Context, u *url.URL) *grpc.ClientConn
...
 			connect.WithDialOptions(DefaultDialOptions(b.generateTokenFunc)...)),
+		nsmgrproxy.WithRegistryConnectOptions(
+			DefaultDialOptions(b.generateTokenFunc)...,
+		),
 	)
-	serveURL := grpcutils.TargetToURL(b.newAddress("nsmgr-proxy"))
 
 	serve(ctx, serveURL, mgr.Register)
-	log.FromContext(ctx).Infof("%v listen on: %v", name, serveURL)
+	log.FromContext(ctx).Infof("%v: %v listen on: %v", b.DNSDomainName, name, serveURL)
```
Comment (edwarnicke) [inline pkg/tools/sandbox/builder.go:313]: Could we make this Debug level?

File: pkg/tools/sandbox/builder.go [hunk 5]
```diff
@@ -339,7 +366,7 @@ func (b *Builder) newNSMgr(ctx context.Context, address string, registryURL *url
 	mgr := b.supplyNSMgr(ctx, generateTokenFunc, options...)
 
 	serve(ctx, serveURL, mgr.Register)
-	log.FromContext(ctx).Infof("%v listen on: %v", nsmgrName, serveURL)
+	log.FromContext(ctx).Infof("%v: %v listen on: %v", b.DNSDomainName, nsmgrName, serveURL)
```
Comment (edwarnicke) [inline pkg/tools/sandbox/builder.go:369]: Could we make this debug level?

File: pkg/tools/sandbox/builder.go [hunk 6]
```diff
@@ -363,14 +390,14 @@ func serve(ctx context.Context, u *url.URL, register func(server *grpc.Server))
...
+func (b *Builder) newRegistryProxy(ctx context.Context) *RegistryEntry {
 	if b.supplyRegistryProxy == nil {
 		return nil
 	}
-	result := b.supplyRegistryProxy(ctx, b.Resolver, b.DNSDomainName, nsmgrProxyURL, DefaultDialOptions(b.generateTokenFunc)...)
+	result := b.supplyRegistryProxy(ctx, b.resolver, DefaultDialOptions(b.generateTokenFunc)...)
 	serveURL := grpcutils.TargetToURL(b.newAddress("reg-proxy"))
 	serve(ctx, serveURL, result.Register)
-	log.FromContext(ctx).Infof("registry-proxy-dns listen on: %v", serveURL)
+	log.FromContext(ctx).Infof("%v: registry-proxy-dns listen on: %v", b.DNSDomainName, serveURL)
```
Comment (edwarnicke) [inline pkg/tools/sandbox/builder.go:400]: Could we make this Debug level?

File: pkg/tools/sandbox/builder.go [hunk 7]
```diff
@@ -384,7 +411,7 @@ func (b *Builder) newRegistry(ctx context.Context, proxyRegistryURL *url.URL) *R
 	result := b.supplyRegistry(ctx, b.registryExpiryDuration, proxyRegistryURL, DefaultDialOptions(b.generateTokenFunc)...)
 	serveURL := grpcutils.TargetToURL(b.newAddress("reg"))
 	serve(ctx, serveURL, result.Register)
-	log.FromContext(ctx).Infof("Registry listen on: %v", serveURL)
+	log.FromContext(ctx).Infof("%v: registry listen on: %v", b.DNSDomainName, serveURL)
```
Comment (edwarnicke) [inline pkg/tools/sandbox/builder.go:414]: Could we make this Debug level?

Comment (denis-tingaikin) [reply pkg/tools/sandbox/builder.go]: Thanks, fixed!

File: pkg/tools/sandbox/builder.go (hunk 6 above)
Comment (denis-tingaikin) [reply pkg/tools/sandbox/builder.go]: Thanks, fixed!

File: pkg/tools/sandbox/builder.go (hunk 5 above)
Comment (denis-tingaikin) [reply pkg/tools/sandbox/builder.go]: Thanks, fixed!

File: pkg/tools/sandbox/builder.go (hunk 4 above)
Comment (denis-tingaikin) [reply pkg/tools/sandbox/builder.go]: Thanks, fixed!

File: pkg/registry/common/dnsresolve/common.go (hunk 3 above)
Comment (denis-tingaikin) [reply pkg/registry/common/dnsresolve/common.go]: Thanks, fixed!

File: pkg/tools/clienturlctx/context.go (hunk 2 above)
Comment (denis-tingaikin) [reply pkg/tools/clienturlctx/context.go]: Thanks, fixed!

**Pull request merged**